### PR TITLE
Add support for Symfony 6

### PR DESCRIPTION
**Pull request by macintoshplus**
I have made these changes to use this bundle on Symfony 6.1 project. But I don't know how to manage correctly this change between version <= 6.0 and >= 6.1.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -24,20 +24,20 @@
...
+        "symfony/dependency-injection": "^4.4 || ^5.4 || ^6.1",
+        "symfony/filesystem": "^4.4 || ^5.4 || ^6.1",
+        "symfony/finder": "^4.4 || ^5.4 || ^6.1",
+        "symfony/form": "^4.4 || ^5.4 || ^6.1",
+        "symfony/framework-bundle": "^4.4 || ^5.4 || ^6.1",
+        "symfony/http-foundation": "^4.4 || ^5.4 || ^6.1",
+        "symfony/http-kernel": "^4.4 || ^5.4 || ^6.1",
+        "symfony/options-resolver": "^4.4 || ^5.4 || ^6.1",
+        "symfony/service-contracts": "^1.1 || ^2.0 || ^3.0",
+        "symfony/translation": "^4.4 || ^5.4 || ^6.1",
```
Comment (loic425) [inline composer.json:39]:
```suggestion
        "symfony/asset": "^5.4 || ^6.0",
        "symfony/config": "^5.4 || ^6.0",
        "symfony/console": "^5.4 || ^6.0",
        "symfony/dependency-injection": "^5.4 || ^6.0",
        "symfony/filesystem": "^5.4 || ^6.0",
        "symfony/finder": "^5.4 || ^6.0",
        "symfony/form": "^4.4 || ^5.4 || ^6.0",
        "symfony/framework-bundle": "^5.4 || ^6.0",
        "symfony/http-foundation": "^5.4 || ^6.0",
        "symfony/http-kernel": "^5.4 || ^6.0",
        "symfony/options-resolver": "^5.4 || ^6.0",
        "symfony/service-contracts": "^1.1 || ^2.0 || ^3.0",
        "symfony/translation": "^5.4 || ^6.0",
```

File: composer.json [hunk 2]
```diff
@@ -46,9 +46,9 @@
         "phpspec/phpspec": "^7.0",
         "phpunit/phpunit": "^9.4",
         "sylius-labs/coding-standard": "^4.0.2",
-        "symfony/browser-kit": "^4.4 || ^5.4",
-        "symfony/security-csrf": "^4.4 || ^5.4",
-        "symfony/twig-bundle": "^4.4 || ^5.4",
+        "symfony/browser-kit": "^4.4 || ^5.4 || ^6.1",
+        "symfony/security-csrf": "^4.4 || ^5.4 || ^6.1",
+        "symfony/twig-bundle": "^4.4 || ^5.4 || ^6.1",
```
Comment (loic425) [inline composer.json:51]:
```suggestion
        "symfony/browser-kit": "^5.4 || ^6.0",
        "symfony/security-csrf": "^5.4 || ^6.0",
        "symfony/twig-bundle": " ^5.4 || ^6.0",
```

**Review by loic425 — CHANGES_REQUESTED** [review]

**Comment by loic425** [comment]
@macintoshplus Please could you update the CI configuration please?

**Comment by macintoshplus** [comment]
> @macintoshplus Please could you update the CI configuration please?

I have change the Symfony version:

* remove Symfony 4.4
* keep Symfony 5.4
* add Symfony 6.0

**Review by loic425 — DISMISSED** [review]

File: src/Filesystem/FilesystemInterface.php [hunk 3]
```diff
@@ -37,7 +37,7 @@ public function copy(string $originFile, string $targetFile, bool $override = fa
      *
      * @throws IOException On any directory creation failure
      */
-    public function mkdir($dirs, int $mode = 0777);
+    public function mkdir(iterable|string $dirs, int $mode = 0777) ;
```
Comment (loic425) [inline src/Filesystem/FilesystemInterface.php:40]: why these changes are necessary?

Comment (macintoshplus) [reply src/Filesystem/FilesystemInterface.php:40]: The class `Symfony\Component\Filesytem\Filesystem` has been changed.
This class is the parent of `Sylius\Bundle\ThemeBundle\Filesystem\Filesystem`. And this class implements this interface.

Comment (loic425) [reply src/Filesystem/FilesystemInterface.php:40]: Indeed, but this is very strange why there is a FilesystemInterface in this package here...
While not using Symfony interface directly or extending it... I don't know this package at all, but if we change this interface, this is indeed a bc-break and we should make a new release to add support for Symfony 6.

@Roshyo What do you think?

Comment (loic425) [reply src/Filesystem/FilesystemInterface.php:40]: @macintoshplus Could you add a note on UPGRADE.md for the interface changes?

Comment (macintoshplus) [reply src/Filesystem/FilesystemInterface.php:40]: @loic425 I have just updated the `Upgrade.md` file.

Comment (alexander-schranz) [reply src/Filesystem/FilesystemInterface.php:40]: Alternative to avoid future break the `Filesystem` implementation could inject the `filesystem` service instead of extending it: https://github.com/Sylius/SyliusThemeBundle/blob/2741cd68a8c056d56a5472445786832637c29704/src/Filesystem/Filesystem.php#L18-L24

This way the interface could stay as it is and as the Filesystem is final nobody extended from it and so no bc break would be required in this interface.

**Comment by lchrusciel** [comment]
Covered already in #121. Thanks a lot for work Jean-Baptiste, and for collaboration over the solution Stephane and Alexandre 🎉
